addons/account/models/account_move.py

```diff
@@ -3227,6 +3233,24 @@ def _get_new_hash(self, secure_seq_number):
         #build and return the hash
         return self._compute_hash(prev_move.inalterable_hash if prev_move else u'')
 
+    @api.model
+    def _get_move_hash_domain(self, common_domain=False):
+        # hash customer invoices/refunds on send, the rest on post
+        out_types = ('out_invoice', 'out_refund')
+        common_domain = common_domain or []
+        return expression.AND(
+            [common_domain, expression.OR([
+                [('is_move_sent', '=', True), ('move_type', 'in', out_types)],
+                [('state', '=', 'posted'), ('move_type', 'not in', out_types)],
```

Comment:
That order makes it more obvious how the two cases are related
```suggestion
                [('move_type', 'in', out_types), ('is_move_sent', '=', True)],
                [('move_type', 'not in', out_types), ('state', '=', 'posted')],
```